spec/controllers/application_records_controller_spec.rb

```diff
@@ -322,6 +323,28 @@
     end
   end
 
+  describe 'POST #toggle_saved_for_later' do
+    before :each do
+      when_current_user_is :staff
+    end
+    context 'record not saved for later' do
+      let!(:record) { create :application_record, saved_for_later: false }
+      let(:submit) { post :toggle_saved_for_later, id: record.id }
+      it 'calls record.save_for_later' do
+        expect_any_instance_of(ApplicationRecord).to receive(:save_for_later)
```

Comment:
Does it not work here to establish a mock on the specific instance you create on line 331?